backend/gsr_booking/views.py

```diff
@@ -275,3 +284,71 @@ def get(self, request):
                 request.user, request.user.booking_groups.filter(name="Penn Labs").first()
             )
         )
+
+
+class GSRShareCodeViewSet(
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    ViewSet for creating, retrieving, and deleting GSR share codes.
+    """
+
+    serializer_class = GSRShareCodeSerializer
+    lookup_field = "code"
+    lookup_value_regex = r"[A-Za-z0-9_-]{8}"
+
+    permission_classes = [IsShareCodeOwner]
+
+    def get_permissions(self):
```

Comment:
nit: looks like you might be able to remove this, since all of this logic is also abstracted already